httpclient5-cache/src/main/java/org/apache/hc/client5/http/impl/cache/ConditionalRequestBuilder.java

```diff
@@ -58,26 +56,17 @@
     public T buildConditionalRequest(final T request, final HttpCacheEntry cacheEntry) {
         final T newRequest = messageCopier.create(request);
 
-        final Header eTag = cacheEntry.getFirstHeader(HeaderConstants.ETAG);
+        final Header eTag = cacheEntry.getFirstHeader(HttpHeaders.ETAG);
         if (eTag != null) {
-            newRequest.setHeader(HeaderConstants.IF_NONE_MATCH, eTag.getValue());
+            newRequest.setHeader(HttpHeaders.IF_NONE_MATCH, eTag.getValue());
         }
-        final Header lastModified = cacheEntry.getFirstHeader(HeaderConstants.LAST_MODIFIED);
+        final Header lastModified = cacheEntry.getFirstHeader(HttpHeaders.LAST_MODIFIED);
         if (lastModified != null) {
-            newRequest.setHeader(HeaderConstants.IF_MODIFIED_SINCE, lastModified.getValue());
+            newRequest.setHeader(HttpHeaders.IF_MODIFIED_SINCE, lastModified.getValue());
         }
-        boolean mustRevalidate = false;
-        final Iterator<HeaderElement> it = MessageSupport.iterate(cacheEntry, HeaderConstants.CACHE_CONTROL);
-        while (it.hasNext()) {
-            final HeaderElement elt = it.next();
-            if (HeaderConstants.CACHE_CONTROL_MUST_REVALIDATE.equalsIgnoreCase(elt.getName())
-                    || HeaderConstants.CACHE_CONTROL_PROXY_REVALIDATE.equalsIgnoreCase(elt.getName())) {
-                mustRevalidate = true;
-                break;
-            }
-        }
-        if (mustRevalidate) {
-            newRequest.addHeader(HeaderConstants.CACHE_CONTROL, HeaderConstants.CACHE_CONTROL_MAX_AGE + "=0");
+        final ResponseCacheControl cacheControl = CacheControlHeaderParser.INSTANCE.parse(cacheEntry);
```

Comment:
Similarly, since we already possess the ResponseCacheControl, we should simply pass it as a parameter